### PR TITLE
RouteSync::onMsg supports ipv6 route

### PR DESCRIPTION
**Pull request by qiluo-msft**

### CODE REVIEW

File: fpmsyncd/routesync.cpp [hunk 1]
```diff
@@ -25,34 +25,32 @@ void RouteSync::onMsg(int nlmsg_type, struct nl_object *obj)
...
-    char ifname[MAX_ADDR_SIZE + 1] = {0};
-    uint32_t ipv4;
-    int prefix;
+    char destipprefix[MAX_ADDR_SIZE + 1] = {0};
 
     dip = rtnl_route_get_dst(route_obj);
-    /* Supports IPv4 address only for now */
-    if (rtnl_route_get_family(route_obj)  != AF_INET)
+    nl_addr2str(dip, destipprefix, MAX_ADDR_SIZE);
+    SWSS_LOG_DEBUG("destipprefix=%s\n", destipprefix);
```
Comment (lguohan) [inline fpmsyncd/routesync.cpp]: not sure we need this debug log, at minimum it needs more be descriptive, like receive new route message  dest ip prefix %s

File: fpmsyncd/routesync.cpp [hunk 2]
```diff
@@ -25,34 +25,32 @@ void RouteSync::onMsg(int nlmsg_type, struct nl_object *obj)
...
-                      __FUNCTION__, ifname, nl_object_get_type(obj));
-        return;
+        case AF_INET:
+        case AF_INET6:
+            break;
+        default:
+            SWSS_LOG_INFO("%s: Unknown route family support: %s (object: %s)\n",
+                          __FUNCTION__, destipprefix, nl_object_get_type(obj));
+            return;
     }
```
Comment (lguohan) [inline fpmsyncd/routesync.cpp]: change this to if (family != AF_NET and family != AF_INET) return. it's more clear then switch.

File: fpmsyncd/routesync.cpp [hunk 3]
```diff
@@ -106,12 +103,13 @@ void RouteSync::onMsg(int nlmsg_type, struct nl_object *obj)
             nexthops += nh.to_string();
         }
 
-        rtnl_link_i2name(m_link_cache, ifindex, ifname, MAX_ADDR_SIZE);
+        rtnl_link_i2name(m_link_cache, ifindex, ifname, IFNAMSIZ);
+        SWSS_LOG_DEBUG("ifname=%s\n", ifname);
```
Comment (lguohan) [inline fpmsyncd/routesync.cpp]: not sure we need this debug log, it's like dev debug log.

**Comment by lguohan** [comment]
:clock1:

File: fpmsyncd/routesync.cpp (hunk 1 above)
Comment (qiluo-msft) [inline fpmsyncd/routesync.cpp]: Resolved.

File: fpmsyncd/routesync.cpp [hunk 4]
```diff
@@ -29,17 +29,14 @@ void RouteSync::onMsg(int nlmsg_type, struct nl_object *obj)
...
     {
-        case AF_INET:
-        case AF_INET6:
-            break;
-        default:
-            SWSS_LOG_INFO("%s: Unknown route family support: %s (object: %s)\n",
-                          __FUNCTION__, destipprefix, nl_object_get_type(obj));
-            return;
+        SWSS_LOG_INFO("%s: Unknown route family support: %s (object: %s)\n",
+                      __FUNCTION__, destipprefix, nl_object_get_type(obj));
```
Comment (kcudnik) [inline fpmsyncd/routesync.cpp]: __FUNCTION__ is always logged in any SWSS_LOG_INFO, no need it here

File: fpmsyncd/routesync.cpp (hunk 2 above)
Comment (qiluo-msft) [inline fpmsyncd/routesync.cpp]: Resolved.

File: fpmsyncd/routesync.cpp (hunk 3 above)
Comment (qiluo-msft) [inline fpmsyncd/routesync.cpp]: Resolved.

File: fpmsyncd/routesync.cpp (hunk 4 above)
Comment (qiluo-msft) [inline fpmsyncd/routesync.cpp]: Resolved.

File: fpmsyncd/routesync.cpp [hunk 5]
```diff
@@ -25,34 +25,27 @@ void RouteSync::onMsg(int nlmsg_type, struct nl_object *obj)
...
-    char ifname[MAX_ADDR_SIZE + 1] = {0};
-    uint32_t ipv4;
-    int prefix;
+    char destipprefix[MAX_ADDR_SIZE + 1] = {0};
 
     dip = rtnl_route_get_dst(route_obj);
-    /* Supports IPv4 address only for now */
-    if (rtnl_route_get_family(route_obj)  != AF_INET)
+    nl_addr2str(dip, destipprefix, MAX_ADDR_SIZE);
+    SWSS_LOG_DEBUG("Receive new route message dest ip prefix: %s\n", destipprefix);
```
Comment (stcheng) [inline fpmsyncd/routesync.cpp]: adding nlmsg_type here to log more information

File: fpmsyncd/routesync.cpp [hunk 6]
```diff
@@ -63,7 +56,7 @@ void RouteSync::onMsg(int nlmsg_type, struct nl_object *obj)
                 std::vector<FieldValueTuple> fvVector;
```
Comment (stcheng) [inline fpmsyncd/routesync.cpp]: remove the std:: and the std:: on line 117

**Comment by stcheng** [comment]
🚢 waiting for resolving all comments and then merge

**Pull request merged**